File: training-data/fortran/floyd_warshall.f90
program floyd_warshall
    implicit none
    integer, parameter :: n = 4, inf = 999999
    integer :: dist(n, n)
    integer :: i, j, k

    dist = inf
    do i = 1, n
        dist(i, i) = 0
    end do
    dist(1, 2) = 3
    dist(1, 3) = 8
    dist(2, 4) = 1
    dist(3, 2) = 4
    dist(4, 1) = 2
    dist(4, 3) = 5

    do k = 1, n
        do i = 1, n
            do j = 1, n
                if (dist(i, k) /= inf .and. dist(k, j) /= inf) then
                    if (dist(i, k) + dist(k, j) < dist(i, j)) then
                        dist(i, j) = dist(i, k) + dist(k, j)
                    end if
                end if
            end do
        end do
    end do

    do i = 1, n
        print *, dist(i, :)
    end do
end program floyd_warshall

File: training-data/nim/floyd_warshall.nim
const INF = 1_000_000

proc floydWarshall(graph: seq[seq[int]]): seq[seq[int]] =
  let n = graph.len
  result = graph
  for k in 0 ..< n:
    for i in 0 ..< n:
      for j in 0 ..< n:
        if result[i][k] + result[k][j] < result[i][j]:
          result[i][j] = result[i][k] + result[k][j]

let graph = @[
  @[0, 5, INF, 10],
  @[INF, 0, 3, INF],
  @[INF, INF, 0, 1],
  @[INF, INF, INF, 0]
]

let dist = floydWarshall(graph)
for row in dist:
  echo row
